src/libsync/pushnotifications.cpp

```diff
@@ -194,4 +205,76 @@ void PushNotifications::handleNotifyActivity()
     qCInfo(lcPushNotifications) << "Push activity arrived";
     emit activitiesChanged(_account);
 }
+void PushNotifications::onWebSocketPongReceived(quint64 /*elapsedTime*/, const QByteArray &payload)
+{
+    handleTimeoutPong(payload);
+}
+
+void PushNotifications::handleTimeoutPong(const QByteArray &payload)
+{
+    // We are not interested in different pongs from the server
+    const auto expectedPingPayload = getTimeoutPingPayload();
+    if (payload != expectedPingPayload) {
+        return;
+    }
+
+    qCDebug(lcPushNotifications) << "Pong received in time";
+
+
+    _pingTimedOutTimer.stop();
```

Comment:
@FlexW You may want to move it into `startPingTimeoutTimer` just to make sure the stop is always called, and you don't need to call it every time you call this method. The same goes for the other timer.